### PR TITLE
Allow callbacks to be removed

### PR DESCRIPTION
**Pull request by quis**
We’ve had a user who’s said:

> Seems configured callbacks cannot be removed once they’re set as the fields have a presence check. Is that intentional?

This means it’s not working as they expect. Rather than have to go and change stuff in the database for them, let’s make it work as they’d expect.

Only lets you clear the form if you remove both the token and the URL.

### CODE REVIEW

**Comment by quis** [comment]
Looking at the API, this needs more work. Admin should send a `DELETE` request if the fields are blank.

**Comment by quis** [comment]
Updated to send a `DELETE` request to the API.

File: app/main/views/api_keys.py [hunk 1]
```diff
@@ -208,17 +208,24 @@ def delivery_status_callback(service_id):
...
                     user_id=current_user.id,
                     callback_api_id=delivery_status_callback.get('id')
                 )
-        else:
+        elif delivery_status_callback and not form.url.data:
+            service_api_client.delete_service_callback_api(
+                service_id,
+                delivery_status_callback['id'],
+            )
+        elif form.url.data:
```
[the remark this answers is not in the document]
Comment (quis) [reply app/main/views/api_keys.py]: Good idea, added in 665b37f

[the remark this answers is not in the document]
Comment (quis) [reply app/main/views/api_keys.py]: That shouldn’t have been an error; changed it to d781a68 instead

**Comment by leohemsted** [comment]
- [ ] https://github.com/alphagov/notifications-api/pull/1933

**Pull request merged**